src/System.Windows.Forms/src/System/Windows/Forms/ListViewLabelEditNativeWindow.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    internal class ListViewLabelEditNativeWindow : NativeWindow
+    {
+        private const uint TextSelectionChanged = 0x8014;
+
+        private ListView _owner;
+
+        private AccessibleObject? _accessibilityObject;
+
+        private User32.WINEVENTPROC? _winEventProc;
+
+        private nint _valueChangeHook;
+        private nint _textSelectionChangedHook;
+
+        private bool _winEventHooksInstalled;
+
+        public ListViewLabelEditNativeWindow(ListView owner)
+        {
+            _owner = owner.OrThrowIfNull();
+        }
+
+        public AccessibleObject AccessibilityObject => _accessibilityObject ??= new ListViewLabelEditAccessibleObject(_owner, this);
+
+        public override void ReleaseHandle()
+        {
+            if (_winEventHooksInstalled)
+            {
+                User32.UnhookWinEvent(_valueChangeHook);
+                User32.UnhookWinEvent(_textSelectionChangedHook);
+
+                _winEventHooksInstalled = false;
+            }
+
+            if (_accessibilityObject is not null)
+            {
+                // When a window that previously returned providers has been destroyed,
+                // you should notify UI Automation by calling the UiaReturnRawElementProvider
+                // as follows: UiaReturnRawElementProvider(hwnd, 0, 0, NULL). This call tells
+                // UI Automation that it can safely remove all map entries that refer to the specified window.
+                UiaCore.UiaReturnRawElementProvider(Handle, 0, 0, null);
+
+                if (OsVersion.IsWindows8OrGreater)
+                {
+                    UiaCore.UiaDisconnectProvider(AccessibilityObject);
+                }
+            }
+
+            base.ReleaseHandle();
+        }
+
+        protected override void OnHandleChange()
+        {
+            base.OnHandleChange();
+
+            if (Handle == IntPtr.Zero)
+            {
+                return;
+            }
+
+            // Install winevent hooks only in the case when the parent listview already has an accessible object created.
+            // If we won't install hooks at the label edit startup then Narrator won't announce the text pattern for it.
+            // If we will just check for UiaClientsAreListening then we will have hooks installed even if Narrator isn't currently run.
```

Comment:
Did I get correctly the intent?
```suggestion
            // Install winevent hooks *only* when the parent listview has an existing accessible object.
            // If we don't install hooks at the label edit startup then assistive tech (e.g. Narrator) won't announce the text pattern for it.
            // By invoking UiaClientsAreListening we will have hooks installed even if the assistive tech isn't currently run.
```

Also "_...then assistive tech (e.g. Narrator) won't announce the text pattern for **it**._" What is "_it_" in this context?

Reply:
Thanks for cleaning up the texts. I assume "it" means the label edit control.